core/src/main/java/com/valtech/aem/saas/core/fulltextsearch/SearchRedirectModelImpl.java

```diff
@@ -0,0 +1,144 @@
+import java.util.stream.Stream;
+import javax.annotation.PostConstruct;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.jcr.resource.api.JcrResourceConstants;
+import org.apache.sling.models.annotations.DefaultInjectionStrategy;
+import org.apache.sling.models.annotations.Exporter;
+import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.injectorspecific.OSGiService;
+import org.apache.sling.models.annotations.injectorspecific.Self;
+import org.apache.sling.models.annotations.injectorspecific.SlingObject;
+import org.apache.sling.models.annotations.injectorspecific.ValueMapValue;
+import org.apache.sling.models.factory.ModelFactory;
+
+/**
+ * Search redirect component sling model that handles component's rendering.
+ */
+@Slf4j
+@Model(adaptables = SlingHttpServletRequest.class,
+    adapters = {SearchRedirectModel.class, ComponentExporter.class},
+    defaultInjectionStrategy = DefaultInjectionStrategy.OPTIONAL,
+    resourceType = RESOURCE_TYPE)
+@Exporter(name = ExporterConstants.SLING_MODEL_EXPORTER_NAME,
+    extensions = ExporterConstants.SLING_MODEL_EXTENSION)
+public class SearchRedirectModelImpl implements SearchRedirectModel {
+
+  public static final String RESOURCE_TYPE = "saas-aem-module/components/searchredirect";
+  private static final String HTML_EXTENSION = ".html";
+
+  @Getter
+  @ValueMapValue
+  private String searchFieldPlaceholderText;
+
+  @Getter
+  private String searchUrl;
+
+  @Getter
+  private String autocompleteUrl;
+
+  @Getter
+  @ValueMapValue(name = JcrResourceConstants.SLING_RESOURCE_TYPE_PROPERTY)
+  private String exportedType;
+
+  @ValueMapValue
+  private String searchPagePath;
+
+  @JsonIgnore
+  @Getter
+  private String configJson;
+
+  @Self
+  private SlingHttpServletRequest request;
+
+  @SlingObject
+  private Resource resource;
+
+  @SlingObject
+  private ResourceResolver resourceResolver;
+
+  @OSGiService
+  private PathTransformer pathTransformer;
+
+  @OSGiService
+  private ModelFactory modelFactory;
+
+  private SearchModel searchModel;
+
+  private Resource searchPageResource;
+
+  @PostConstruct
+  private void init() {
+    getSearchPageResource().ifPresent(r -> searchPageResource = r);
+    getSearchModel().ifPresent(search -> searchModel = search);
+    retrieveAutocompleteUrl().ifPresent(url -> autocompleteUrl = url);
+    resolveSearchFieldPlaceholderText().ifPresent(s -> searchFieldPlaceholderText = s);
+    createSearchPageUrl().ifPresent(s -> searchUrl = s);
+    configJson = getSearchConfigJson();
+  }
+
+  private Optional<String> createSearchPageUrl() {
+    return Optional.ofNullable(searchPageResource)
+        .map(r -> pathTransformer.map(request, r.getPath()))
+        .map(s -> s + HTML_EXTENSION);
```

Comment:
Adding HTML should be part of the path transformer. There are AEM's around which don't have HTML extension

Reply:
`PathTransformer::map` is used. This only maps the resourcePath (not adding extension).
Probably a better option is the `PathTransformer::externalize`, but this method returns a list of string.
Should I always get the first item from the list?

Reply:
I think we just need to work a bit on naming. The word "externalize" is used in AEM's externalizer and it is adding the extension. The word map is used in slings resource resolver and doesn't add the extension.
None of these ootb methods are returning a list of paths. 
In our module I don't see the usecase for using the resourceResolver.map. Do you?

As a suggestion we could rename 
`PathTransformer::map` to `PathTransformer::externalize` and we rename `PathTransformer::externalize` to `PathTransformer::externalizeList`

Reply:
We use PathTransformer::map in `SearchTabModelImpl::constructJsonExportUrl` and `SearchModelImpl::createAutosuggestUrl`
Here we only need resource path mapping. IMO method names are ok. Maybe we need to add additional `PathTransformer::externalizeList` (that returns a list of strings) and make `PathTransformer::externalize` to return String

Reply:
Ok that sounds good. So the map will never an extension and the externalize the / extenalizeList will add it by default. 
Please adjust than the Javadoc accordingly which describes this